src/marqo/tensor_search/health.py

```diff
@@ -0,0 +1,57 @@
+from typing import Optional, Tuple
+import copy
+
+from marqo.config import Config
+from marqo import errors
+from marqo._httprequests import HttpRequests
+from marqo.tensor_search.enums import HealthStatuses
+
+
+def generate_heath_check_response(config: Config, index_name: Optional[str] = None) -> dict:
+    """Generate the health check response for check_heath(), check_index_health() APIs in tensor_search"""
+    marqo_status = get_marqo_status()
+    marqo_os_status = get_marqo_os_status(config, index_name=index_name)
+    marqo_status, marqo_os_status = aggregate_status(marqo_status, marqo_os_status)
+
+    return {
+        "status": marqo_status,
+        "backend": {
+            "status": marqo_os_status
+        }
+    }
+
+
+def get_marqo_status() -> str:
+    """Check the Marqo instance status."""
+    return HealthStatuses.green
+
+
+def get_marqo_os_status(config: Config, index_name: Optional[str] = None) -> str:
+    """Check the Marqo-os backend status."""
+    TIMEOUT = 3
+    marqo_os_health_check_response = None
+    path = f"_cluster/health/{index_name}" if index_name else "_cluster/health"
+
+    try:
+        timeout_config = copy.deepcopy(config)
+        timeout_config.timeout = TIMEOUT
+        marqo_os_health_check_response = HttpRequests(timeout_config).get(path=path)
+    except errors.BackendCommunicationError:
```

Comment:
We may also need to handle timeout errors : https://github.com/marqo-ai/py-marqo/blob/d22fe41e9cf7efb10ef1a9a3ed055746612041a2/src/marqo/_httprequests.py#L77C34-L77C34 

or internal errors, which is the super class of these errors : https://github.com/marqo-ai/py-marqo/blob/d22fe41e9cf7efb10ef1a9a3ed055746612041a2/src/marqo/errors.py#L145C15-L145C15